aeron-driver/src/main/c/aeron_driver_common.h

```diff
@@ -67,6 +67,20 @@ typedef struct aeron_tetherable_position_stct
 }
 aeron_tetherable_position_t;
 
+typedef void (*aeron_untethered_subscription_state_change_func_t)(
+        aeron_tetherable_position_t *,
+        int64_t now_ns,
+        aeron_subscription_tether_state_t new_state,
+        int32_t stream_id,
+        int32_t session_id);
+
+void aeron_untethered_subscription_state_change(
+        aeron_tetherable_position_t *tetherable_position,
+        int64_t now_ns,
+        aeron_subscription_tether_state_t new_state,
+        int32_t stream_id,
+        int32_t session_id);
+
 typedef struct aeron_subscribable_stct
```

Comment:
This is more a general design direction question (perhaps for @tmontgomery and @mjpt777).  Would there be a way to add the agent logging without needed function pointers at each place we need to log?  I tend to find the function pointers can be hard to follow.  The benefit of agent based logging in Java is that the code doesn't change (i.e. doesn't get cluttered with log statements) and the performance impact is negligible.  However in C introducing a function pointer just for logging adds both cognitive and performance overhead.

Reply:
I agree. However, for hooking, we can't LD_PRELOAD on functions that are not entry points. Or we could do that as was done previously with sendmsg/recvmsg/sendmmsg/recvmmsg.